test/js/api/domain/gather-files-spec.js

```diff
@@ -6,55 +7,59 @@ var assert = require('chai').assert,
+        return gatherFiles(path.normalize(fakePath), options || {});    
     };
-        
-    it('should return all the files if there is not a matching filename', function() {
-        var expected = [
-            {"name":"a.js","file":"a.js"},
-            {"name":"b.js","file":"b.js"},
-            {"name":"c.js","file":"c.js"}];
-        
-        var actual = getFiles('/fake/dir', ['a.js', 'b.js', 'c.js']);
-        assert.deepEqual(actual, expected);
-    });
 
-    it('should return only the files whose name won\'t match the pattern', function() {
-        var expected = [{"name":"a.js","file":"a.js"}, {"name":"c.js","file":"c.js"}];        
-        var actual = getFiles('/fake/dir', ['a.js', 'c.js', 'response_b.js']);
-        assert.deepEqual(actual, expected);
+    it('should recognize responses as files starting with \'response_\' by default', function() {
+        assert.deepEqual(
+            _test(_getFiles(['request_a', 'response_a', 'request_b', 'res_b'])), 
+            _expected(['request_a', 'request_b', 'res_b']));
 
     });    
     
-    it('shouldn\'t return anything if all of the files match the pattern', function() {
-        var actual = getFiles('/fake/dir', ['response_a.js', 'response_b.js']);
-        assert.deepEqual(actual, []);
-    });    
+    it('should only return saved requests by default', function() {
+        assert.deepEqual(
+            _test(_getFiles(['request_a', 'response_a', 'request_b', 'response_b'])), 
+            _expected(['request_a', 'request_b']));
 
-    it('should respect the provided response pattern', function() {
-        var expected = [
-            {"name":"a.xml","file":"a.xml"},
-            {"name":"c.txt","file":"c.txt"}]; 
-        
-        var actual = getFiles('/fake/dir', ['a.xml', 'b.js', 'c.txt'], 
-                              {patterns: {response: 'b'}});
-        assert.deepEqual(actual, expected);
     });    
     
-    it('should use a default value of \'_response\' if no response pattern is provided', function() {
-        var expected = [{"name":"c.js","file":"c.js"}];        
-        var actual = getFiles('/fake/dir', ['response_a.js', 'c.js', 'response_b.js']);
-        assert.deepEqual(actual, expected);
-
+    it('shouldn\'t return anything if there are only saved responses available, by default', function() {
+        assert.deepEqual(_test(['response_a', 'response_b']), []);
     });    
-    
+
+    it('should be possible to override the default response pattern', function() {
+        assert.deepEqual(
+            _test(_getFiles(['request_a1', 'response_a1', 'request_a2', 'response_a2', 'request_b', 'response_b']),
+                  {patterns: {response: 'response_a'}}), 
+            _expected(['request_a1', 'request_a2', 'request_b', 'response_b']));
+    });            
+
+    it('should be possible to flip the response pattern to filter out requests instead of responses', function() {
```

Comment:
[1-3] yes

that depends on what module you use. The file name has the request.

With the proxy/simple module the file name is just the query string from the GET while in the proxy/params it uses "template(attribute list).ext"

gatherFiles doesn't care about the format for GET files, it just lists all files in the folder. Only for POST files it filters out those that are not "request_"